### PR TITLE
fix(search): AVT1 bugs

### LINKED ISSUE #1363: ♿Search AVT1 - August 2018♿
## Environment
Chrome Version 70.0.3538.67 (Official Build) (64-bit)
macOS Mojave 10/Windows 10

### General
The id search-input-label-2 specified for WAI-ARIA property aria-labelledby on element input is not valid.
screen shot 2018-10-30 at 1 52 42 am

### Small
Multiple WAI-ARIA 'search' landmarks must have unique labels specified with aria-label or aria-labelledby.
screen shot 2018-10-30 at 1 58 07 am
The WAI-ARIA role(s) and/or attribute(s) search are not valid for the element input. See rule help for guidance.
screen shot 2018-10-30 at 1 59 21 am

## How to reproduce
1. [Download the Dynamic Accessibility Plugin tool here](https://ibm.ent.box.com/notes/335005856253?s=mhpgjlgss354mcidsv6rvpuu0bxlmfm4)
2. Run the extension using the August 2018 ruleset

### PR DESCRIPTION
**Pull request by dakahn**
Closes #1363

Implements various changes to address AVT1 bugs found in the search component during testing.

### CODE REVIEW

**Review by asudoh — APPROVED** [review]
LGTM :+1: - Thanks @dakahn!

File: src/components/search/search.hbs [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {{#is suffix "sm"}}
 <div class="bx--form-item">{{/is}}
-  <div data-search class="bx--search bx--search--{{suffix}}{{#if light}} bx--search--light{{/if}}" role="search">
+  <div data-search role="search" class="bx--search bx--search--{{suffix}}{{#if light}} bx--search--light{{/if}}">
     <label id="search-input-label-1" class="bx--label" for="search__input-1">Search</label>
-    <input class="bx--search-input" type="text" role="search" id="search__input-1" placeholder="Search" aria-labelledby="search-input-label-2">
+    <input class="bx--search-input" type="text" id="search__input-1" placeholder="Search" aria-labelledby="search-input-label-1">
```
Comment (joshblack) [inline src/components/search/search.hbs]: This is more-so an issue with the originally markup, but why do we have `label` with `for` and `aria-labelledby` set?

**Review by joshblack — CHANGES_REQUESTED** [review]

File: src/components/search/search.hbs (hunk 1 above)
Comment (dakahn) [reply src/components/search/search.hbs]: Totally. Took it out. 👍

**Review by alisonjoseph — APPROVED** [review]
LGTM

**Pull request merged**